app/uk/gov/hmrc/helptosave/models/account/Transactions.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.models.account
+
+import java.time.LocalDate
+
+import cats.Eq
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.instances.list._
+import cats.instances.string._
+import cats.syntax.apply._
+import cats.syntax.eq._
+import cats.syntax.traverse._
+import play.api.libs.json.{JsString, JsValue, Json, Writes}
+
+sealed trait Operation
+case object Credit extends Operation
+case object Debit extends Operation
+
+object Operation {
+  implicit val eqOperation: Eq[Operation] = Eq.fromUniversalEquals
+  implicit val writes: Writes[Operation] = new Writes[Operation] {
+    override def writes(o: Operation): JsValue = JsString(o match {
+      case Credit ⇒ "credit"
+      case Debit  ⇒ "debit"
+    })
+  }
+}
+
+case class ValidNsiTransaction(
+    sequence:             Int,
+    operation:            Operation,
+    amount:               BigDecimal,
+    transactionDate:      LocalDate,
+    accountingDate:       LocalDate,
+    description:          String,
+    transactionReference: String
+)
+
+object ValidNsiTransaction {
+  private type ValidOrErrorString[A] = ValidatedNel[String, A]
+
+  def apply(nsiTransaction: NsiTransaction): ValidatedNel[String, ValidNsiTransaction] = {
+      def operationValidation(nsiOperation: String): ValidOrErrorString[Operation] =
+        if (nsiOperation === "C") { Valid(Credit) }
+        else if (nsiOperation === "D") { Valid(Debit) }
+        else { Invalid(NonEmptyList.one(s"""Unknown value for operation: "$nsiOperation"""")) }
+
+      def sequenceValidation(nsiSequence: String): ValidOrErrorString[Int] =
+        try Valid(nsiSequence.toInt)
+        catch {
+          case _: NumberFormatException ⇒ Invalid(NonEmptyList.one(s"""Can't parse sequence value as an integer: "$nsiSequence""""))
+        }
+
+    (operationValidation(nsiTransaction.operation), sequenceValidation(nsiTransaction.sequence)).mapN {
```

Comment:
The reason we were trimming the account closed flag was because both `""` and `" "` are valid values. Those aren't valid values of `operation` so that reason for trimming doesn't apply here.